File: app/src/main/java/com/example/firabasefirstexperience/StartActivity.java
package com.example.firabasefirstexperience;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.firebase.auth.FirebaseUser;

public class StartActivity extends AppCompatActivity {



    private static final String TAG = "StartActivity";

    private Button butRegister, butLog;
    private FirebaseUser firebaseUser;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.start_main);

        if(firebaseUser != null) {
            Intent intent = new Intent(StartActivity.this, MainActivity.class);
            startActivity(intent);
            finish();
        }

        butRegister = (Button) findViewById(R.id.butRegister);
        butLog = (Button) findViewById(R.id.butLog);

        butLog.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(StartActivity.this, LoginActivity.class));
            }
        });

        butRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(StartActivity.this, RegisterAcitivty.class));
            }
        });
    }
}

File: app/src/main/java/com/example/firabasefirstexperience/kotlin/StartActivity.kt
package com.example.firabasefirstexperience.kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.example.firabasefirstexperience.R
import com.example.firabasefirstexperience.java.activity.LoginActivity

class StartActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.start_main)


        val butRegister = findViewById<Button>(R.id.butRegister)
        val butLog = findViewById<Button>(R.id.butLog)

        butRegister.setOnClickListener {
            val intent = Intent(this@StartActivity, RegisterActivity::class.java)
            startActivity(intent)
        }

        butLog.setOnClickListener {
            val intent = Intent(this@StartActivity, com.example.firabasefirstexperience.kotlin.LoginActivity::class.java)
            startActivity(intent)
        }
    }
}